### PR TITLE
Ivan/attendance models

### PR DESCRIPTION
**Pull request by IvanYu327**
## Notion ticket link

[Create Attendance initial schema/model](https://www.notion.so/uwblueprintexecs/Create-Attendance-initial-schema-model-1e496cb340eb4651982a5a64d81333ea)

## Implementation description
* Create new models for the attendance sheet and the attendance records on it

## Steps to test
1. 

## What should reviewers focus on?
* new models `attendance_sheets.py` and `attendance_records.py`

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by bowenzhu1 — APPROVED** [review]
this looks good! 

before merging, I was wondering if the migration command worked? did it generate a migration file anywhere?

**Comment by IvanYu327** [comment]
> this looks good!
> 
> before merging, I was wondering if the migration command worked? did it generate a migration file anywhere?

Yeah, it made a file under `versions`

**Comment by bowenzhu1** [comment]
> > this looks good!
> > before merging, I was wondering if the migration command worked? did it generate a migration file anywhere?
> > image
> 
> Yeah @bowenzhu1

I think you can add the migration file to this PR, and we should probably do it for any model changes. pls request another review from me once you do just so i can take a look

thanks!

File: backend/python/app/models/__init__.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 def init_app(app):
     from .access_type import AccessType
+    from .attendance_records import AttendanceRecords
+    from .attendence_sheets import AttendanceSheets
```
Comment (bowenzhu1) [inline backend/python/app/models/__init__.py:11]: noticed some small fixes to make: typo here attendance

File: backend/python/app/models/attendance_records.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from . import db
+from .base_mixin import BaseMixin
+
+supervision_enum = db.Enum("FULL", "PARTIAL", "UNSUPERVISED", name="supervision")
+
+attendance_enum = db.Enum("PRESENT", "CANCELLED", "NO_SHOW", name="attendace")
```
Comment (bowenzhu1) [inline backend/python/app/models/attendance_records.py:6]: typo at the end here as well

**Review by bowenzhu1 — CHANGES_REQUESTED** [review]
some typos I just noticed, could you also add the migration file and then we can merge this?

**Review by bowenzhu1 — COMMENTED** [review]
hm im not sure why the linter is failing, it's giving errors for :
`would reformat /home/runner/work/childrens-aid-society/childrens-aid-society/backend/python/migrations/versions/2022-07-03_57f14d8ff8bf_condense_goals_and_concerns_models.py
would reformat /home/runner/work/childrens-aid-society/childrens-aid-society/backend/python/tests/functional/test_access_type_service.py
would reformat /home/runner/work/childrens-aid-society/childrens-aid-society/backend/python/tests/functional/test_branch_service.py`

but those aren't even in this PR, lmk if u can fix it or if u get stuck on this linting issue

**Comment by IvanYu327** [comment]
>  @bowenzhu1 Not sure how to fix it, the linter is fine on my local repository and I don't see these changes that I'm supposed to make.



> hm im not sure why the linter is failing, it's giving errors for : `would reformat /home/runner/work/childrens-aid-society/childrens-aid-society/backend/python/migrations/versions/2022-07-03_57f14d8ff8bf_condense_goals_and_concerns_models.py would reformat /home/runner/work/childrens-aid-society/childrens-aid-society/backend/python/tests/functional/test_access_type_service.py would reformat /home/runner/work/childrens-aid-society/childrens-aid-society/backend/python/tests/functional/test_branch_service.py`
> 
> but those aren't even in this PR, lmk if u can fix it or if u get stuck on this linting issue

File: backend/python/app/models/attendance_sheets.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+class AttendanceSheets(db.Model, BaseMixin):
+    __tablename__ = "attendance_sheets"
+
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    intake_id = db.Column(db.Integer, db.ForeignKey("intakes.id"), nullable=True)
+    family_name = db.Column(db.String, nullable=False)
+    month = db.Column(month_enum, nullable=False)
+    cic = db.Column(db.String, nullable=False)
```
Comment (bowenzhu1) [inline backend/python/app/models/attendance_sheets.py:28]: got some new requested updates/changes from client:

- Change CIC to CSW (children’s service worker name, also in referral process)
- Change RIV to FCC (foster care coordinator)
- Change Place to Location
- Change attending_parent to attending_family -> We can keep the enums for now of Mom Dad, but we'll probably change this in the future
- Change Access Worker to Child Family Support Worker

Comment (bowenzhu1) [reply backend/python/app/models/attendance_sheets.py:28]: also checkout https://github.com/uwblueprint/childrens-aid-society/pull/149 for the linter issue

Comment (IvanYu327) [reply backend/python/app/models/attendance_sheets.py]: @bowenzhu1 Made those changes and linter passed, but the firebase deploy frontend is failing. I didn't even touch the frontend here lmao.

**Review by bowenzhu1 — APPROVED** [review]
tests look good now, should be able to merge

File: backend/python/app/models/attendance_records.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    date = db.Column(db.String, nullable=False)
+    start_time = db.Column(db.String, nullable=False)
+    end_time = db.Column(db.String, nullable=False)
+    location = db.Column(db.String, nullable=False)
+    attendance = db.Column(attendance_enum, nullable=False)
+    attending_parent = db.Column(attending_parent_enum, nullable=False)
+    staff_transport_time_min = db.Column(db.Integer, nullable=True)
+    driver_transport_time_min = db.Column(db.Integer, nullable=True)
+    foster_parent_transport_time_min = db.Column(db.Integer, nullable=True)
+    child_family_suypport_worker_id = db.Column(
```
Comment (bowenzhu1) [inline backend/python/app/models/attendance_records.py:28]: wait typo here

**Review by bowenzhu1 — CHANGES_REQUESTED** [review]

**Review by bowenzhu1 — APPROVED** [review]

**Pull request merged**